src/Plugin.Maui.CalendarStore/CalendarStore.android.cs

```diff
@@ -377,12 +389,252 @@ public async Task UpdateEvent(string eventId, string title, string description,
 			throw new CalendarStoreException(
 				"There was an error updating the event.");
 		}
+
+		RemoveAllReminders(platformEventId);
+
+		if (reminders is not null)
+		{
+			AddMultipleReminders(platformEventId, reminders);
+		}
+	}
+	//void RemoveAllReminders(long eventId)
+	public async Task UpdateEventWithReminder(string eventId, string title, string description,
+		string location, DateTimeOffset startDateTime, DateTimeOffset endDateTime, bool isAllDay, int reminderMinutes)
+	{
+		await EnsureWriteCalendarPermission();
+
+		var selection = $"{CalendarContract.Events.InterfaceConsts.Id} = ?";
+		var selectionArgs = new[] { eventId };
+
+		using var cursor = platformContentResolver.Query(
+			eventsTableUri, calendarColumns.ToArray(), selection, selectionArgs, null)
+			?? throw new CalendarStoreException("Error while querying events");
+
+		if (!cursor.MoveToFirst())
+		{
+			throw new CalendarStoreException($"Event with ID {eventId} was not found.");
+		}
+
+		long platformEventId = cursor.GetLong(cursor.GetColumnIndex(calendarColumns[0]));
+
+		ContentValues eventToUpdate = new();
+		eventToUpdate.Put(CalendarContract.Events.InterfaceConsts.Dtstart,
+			startDateTime.ToUnixTimeMilliseconds());
+		eventToUpdate.Put(CalendarContract.Events.InterfaceConsts.Dtend,
+			endDateTime.ToUnixTimeMilliseconds());
+		eventToUpdate.Put(CalendarContract.Events.InterfaceConsts.AllDay, isAllDay);
+		eventToUpdate.Put(CalendarContract.Events.InterfaceConsts.Title, title);
+		eventToUpdate.Put(CalendarContract.Events.InterfaceConsts.Description, description);
+		eventToUpdate.Put(CalendarContract.Events.InterfaceConsts.EventLocation, location);
+
+		var updateCount = platformContentResolver?.Update(
+			ContentUris.WithAppendedId(eventsTableUri, platformEventId), eventToUpdate, null, null);
+
+		if (updateCount != 1)
+		{
+			throw new CalendarStoreException($"Failed to update event with ID {eventId}.");
+		}
+
+		UpdateOrAddReminder(platformEventId, reminderMinutes);
+	}
+
+	void UpdateOrAddReminder(long eventId, int reminderMinutes)
+	{
+		var reminderSelection = $"{CalendarContract.Reminders.InterfaceConsts.EventId} = ?";
+		var reminderSelectionArgs = new[] { eventId.ToString() };
+
+		platformContentResolver?.Delete(CalendarContract.Reminders.ContentUri, reminderSelection, reminderSelectionArgs);
 	}
 
 	/// <inheritdoc/>
 	public Task UpdateEvent(CalendarEvent eventToUpdate) =>
 		UpdateEvent(eventToUpdate.Id, eventToUpdate.Title, eventToUpdate.Description,
-			eventToUpdate.Location, eventToUpdate.StartDate, eventToUpdate.EndDate, eventToUpdate.IsAllDay);
+			eventToUpdate.Location, eventToUpdate.StartDate, eventToUpdate.EndDate, eventToUpdate.IsAllDay, eventToUpdate.Reminders.ToArray());
+
+	/// <inheritdoc/>
+	void AddMultipleReminders(long eventId, Reminder[]? reminders)
```

Comment:
Let's name this `AddReminders`, the multiple is already implied by the fact that it takes an array.

Also `/// <inheritdoc/>` in this case doesn't do anything

Reply:
Done.